### PR TITLE
Algorithms report

### PR DESCRIPTION
**Pull request by QijiangYao**
Hi, I'm Qijiang Yao (ID: 45104365), this is my pull request.

### CODE REVIEW

File: image/Code/histogram.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+from skimage import data, exposure, img_as_float
```
Comment (shakes76) [inline image/Code/histogram.py]: Always good to add a header block. Block would include authorship, license, copyright, dates and description of what the driver/module does.

File: image/Code/main.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+from histogram import *
+image = img_as_ubyte(data.camera())  # Convert to 8-bit uint.
+hist, hist_centers = histogram(image, nbins=256, source_range='image', normalize=False)
+
+fig, ax = plt.subplots(ncols=2, figsize=(10, 5))
```
Comment (shakes76) [inline image/Code/main.py]: Inline comments would be good

File: image/README.md [hunk 3]
```diff
@@ -1,14 +1,11 @@
-# Imaging
```
Comment (shakes76) [inline image/README.md]: Should be in own folder for your algorithm. Dependencies list missing

File: image/Code/histogram.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  Args:
+    image(8-bit uint): Input image
+    nbins(Int): Number of bins in histogram
+    source_range(str): 'image’ (default) determines the range from the input image
+    normalize(bool): If True, normalize the histogram by the sum of its values.
+  '''
+  with tf.Session() as sess:
+    image = tf.convert_to_tensor(image, dtype = tf.float32) # Change image to tensorflow
+    sh = sess.run(tf.shape(image))
+    if len(sh) == 3 and sh[-1] < 4:
```
Comment (shakes76) [inline image/Code/histogram.py]: Good to add inline comments even if original code didn't have them

**Comment by shakes76** [comment]
## TF Usage
Adequate usage
## Good Practice (Design/Commenting)
OK
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage 
Module present
## Commit Log
Meaningful commit messages 
Progressive commits used
## Documentation
Description and Comments OK
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
Feedback required, file locations incorrect, requires moving.
Request Description minimal. In the future, a more detailed pull request could've been done. When developer receiving the request is not known, the request could be rejected. Remember you could use/reuse the README info.

**Comment by QijiangYao** [comment]
Hi, Shakes. I have already made the change for my file based on your command.

**Pull request merged**